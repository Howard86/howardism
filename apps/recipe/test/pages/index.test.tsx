import React from "react";
import HomePage from "@/pages/index";
import { render } from "../testUtils";

describe("HomePage", () => {
  it("should render", () => {
    render(<HomePage />, {});
  });
});
