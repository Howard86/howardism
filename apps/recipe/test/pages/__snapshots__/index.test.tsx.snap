// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page matches snapshot 1`] = `
<DocumentFragment>
  <nav
    class="css-nskixn"
  >
    <a
      class="chakra-link css-132n7x6"
      href="/"
    >
      Recipe
    </a>
    <button
      aria-label="open navigation"
      class="chakra-button css-59bxhr"
      type="button"
    >
      <svg
        aria-hidden="true"
        fill="currentColor"
        focusable="false"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 512 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M64 384h384v-42.666H64V384zm0-106.666h384v-42.667H64v42.667zM64 128v42.665h384V128H64z"
        />
      </svg>
    </button>
  </nav>
  <div
    class="chakra-container css-36gip5"
  >
    <div
      class="css-1j2s46t"
    >
      <div
        class="css-i2bp9e"
      />
      <div
        style="display: block; overflow: hidden; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; margin: 0px;"
      >
        <noscript />
        <img
          alt="Landing page background"
          class="css-ar0kck"
          decoding="async"
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
          style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
        />
      </div>
      <div
        class="css-1lp5fbo"
        spacing="10"
      >
        <div
          class="css-0"
        >
          <h1
            class="chakra-heading css-12d46le"
          >
            Check the BEST Recipe
          </h1>
        </div>
        <div
          class="css-0"
        >
          <div
            style="display: inline-block; max-width: 100%; overflow: hidden; position: relative; box-sizing: border-box; margin: 0px;"
          >
            <div
              style="box-sizing: border-box; display: block; max-width: 100%;"
            >
              <img
                alt=""
                aria-hidden="true"
                role="presentation"
                src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMzIwIiBoZWlnaHQ9IjIxOCIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIiB2ZXJzaW9uPSIxLjEiLz4="
                style="max-width: 100%; display: block; margin: 0px; padding: 0px;"
              />
            </div>
            <noscript />
            <img
              class="css-jxmarw"
              decoding="async"
              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
              style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
            />
          </div>
        </div>
        <div
          class="css-76ahhv"
        >
          <h2
            class="chakra-heading css-pvm5ii"
          >
            SHARE YOURS, TOO!
          </h2>
          <p
            class="chakra-text css-34lzpl"
          >
            This is a recipe collection for home-made goodies
          </p>
          <button
            class="chakra-button css-16tb3pm"
            type="button"
          >
            LEARN MORE
          </button>
        </div>
      </div>
    </div>
    <div
      class="css-1glkkdp"
    >
      <h2
        class="chakra-heading css-pvm5ii"
      >
        Top Recipes
      </h2>
      <div
        class="chakra-stack css-rdrbk7"
      >
        <article
          class="chakra-linkbox css-1xvlhys"
        >
          <time
            class="css-saxzvv"
            datetime="Sat Apr 17 2021 21:12:59 GMT+0800 (Taipei Standard Time)"
          >
            New!
          </time>
          <div
            class="chakra-aspect-ratio css-1m7tg19"
          >
            <div
              class="chakra-skeleton css-700nit"
            >
              <img
                alt="recipe photo of Braised Pork Rice"
                class="chakra-image css-1phd9a0"
                src="https://bit.ly/dan-abramov"
              />
            </div>
          </div>
          <h3
            class="chakra-heading css-v4n7kh"
          >
            <a
              class="chakra-linkbox__overlay css-1hnz6hu"
              href="/recipe/1"
            >
              Braised Pork Rice
            </a>
          </h3>
          <p
            class="chakra-text css-0"
          >
            This is an amazing braised pork rice to follow!
          </p>
        </article>
        <article
          class="chakra-linkbox css-1xvlhys"
        >
          <time
            class="css-saxzvv"
            datetime="Sat Apr 17 2021 21:12:59 GMT+0800 (Taipei Standard Time)"
          >
            New!
          </time>
          <div
            class="chakra-aspect-ratio css-1m7tg19"
          >
            <div
              class="chakra-skeleton css-700nit"
            >
              <img
                alt="recipe photo of Braised Pork Rice"
                class="chakra-image css-1phd9a0"
                src="https://bit.ly/dan-abramov"
              />
            </div>
          </div>
          <h3
            class="chakra-heading css-v4n7kh"
          >
            <a
              class="chakra-linkbox__overlay css-1hnz6hu"
              href="/recipe/1"
            >
              Braised Pork Rice
            </a>
          </h3>
          <p
            class="chakra-text css-0"
          >
            This is an amazing braised pork rice to follow!
          </p>
        </article>
        <div
          class="chakra-linkbox css-l2bxy4"
        >
          <h3
            class="chakra-heading css-1389ja7"
          >
            <a
              class="chakra-linkbox__overlay css-1hnz6hu"
              href="/create"
            >
              <svg
                class="chakra-icon css-13otjrl"
                fill="none"
                focusable="false"
                height="1em"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M4 12v8a2 2 0 0 0 2 2h12a2 2 0 0 0 2-2v-8"
                />
                <polyline
                  points="16 6 12 2 8 6"
                />
                <line
                  x1="12"
                  x2="12"
                  y1="2"
                  y2="15"
                />
              </svg>
               Share yours here!
            </a>
          </h3>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
